### ISSUE TITLE
no result for select from function mysql 

### ISSUE THREAD

**Issue by onine007** [ClickHouse/ClickHouse#7885]
but has result for select from table mysql

**How to reproduce**
* Which ClickHouse server version to use
19.15.5
* SQL: `select * from mysql('10.0.4.128:3306', '**', '***', '*****', '****') ;`   

Ok.

0 rows in set. Elapsed: 0.015 sec. 

   table MySQL : 
```
CREATE TABLE druid
(
    `id` String
)
ENGINE = MySQL('10.0.4.128:3306', '**', '***', '******', '********')
        
SELECT count()
FROM druid

┌─count()─┐
│      53 │
└─────────┘

1 rows in set. Elapsed: 0.004 sec. 
```

**Comment by filimonov** [comment]
1) What is MySQL version? distribution ( mysql, mariadb, percona)?
2) can you please show also how table was created on mysql side? (i.e. create table statement)

**Comment by tmpbin** [comment]
I got  same issue,
I tested and used these docker images on docker: `yandex/clickhouse-server:19.15`, `yandex/clickhouse-server:19.16`, `yandex/clickhouse-server:19.17`
 when I run `select count(*) mysql('xxxx', '**', '***', '*****', '****') ` server returns
```
┌─count()─┐
│       0 │
└─────────┘

1 rows in set. Elapsed: 0.051 sec.
```


``` 
select * mysql('xxxx', '**', '***', '*****', '****') 

Ok.

0 rows in set. Elapsed: 0.049 sec.
```

But the following version works fine:
`yandex/clickhouse-server: 19.5`, `yandex/clickhouse-server:last `(19.17.3.7).

```
`select count(*) mysql('xxxx', '**', '***', '*****', '****') ` 

┌─count()─┐
│      30 │
└─────────┘

1 rows in set. Elapsed: 0.035 sec.
```


- Docker Mysql version: `mysql:5.7`

**Comment by tavplubix** [comment]
Probably related: #7762

**Comment by Slach** [comment]
I got the same issue
MySQL 5.1 RedHat Enterprise Linux,
ClickHouse 19.16 rpm installed from Altinity repo

**Comment by alexey-milovidov** [comment]
It's most likely fixed in #7762
